### ISSUE TITLE
[BUG]: MongoDB authentication source not working

### ISSUE THREAD

**Issue by SSTentacleSS** [azukaar/Cosmos-Server#111]
### What happened?

When I tried to create a separate user on the cosmos database to differentiate access - cosmos did not process the mongo uri correctly, it seemed to not like the mongo://.../cosmos part

### What should have happened?

Cosmos must log in to the cosmos database

### How to reproduce the bug?

Execute using mongo shell

```js
use("cosmos")

db.createUser({
  user: "cosmos",
  pwd: "test",
  roles: ["dbOwner"]
})
```

And then set connection string to `mongodb://cosmos:test@address:27017/cosmos` with replacing address to your docker container name/mongo address

### Relevant log output

```log
[ERROR] Database Connect : connection() error occurred during connection handshake: auth error: sasl conversation error: unable to authenticate using mechanism "SCRAM-SHA-1": (InvalidNamespace) Invalid database name: 'cosmos/'
```

### Possible solution

Perhaps it’s worth changing [db.go](https://github.com/azukaar/Cosmos-Server/blob/4d5a148278eaa8e0b61c7141d347cdc8bcf901ae/src/utils/db.go#L20) and replacing string concatenation with an object-based option for changing parameters?

Maybe it's worth doing something like

```go
...
opts := options.Client().ApplyURI(GetBaseMainConfig().MongoDB).SetRetryWrites(true).SetWriteConcern("majority")
client, err = mongo.Connect(context.TODO(), opts)
...
```

[SetWriteConcern](https://github.com/mongodb/mongo-go-driver/blob/1066e2a3b56f6ef4ee0a99140ed8814c9d455de9/mongo/options/clientoptions.go#L894)
[SetRetryWrites](https://github.com/mongodb/mongo-go-driver/blob/1066e2a3b56f6ef4ee0a99140ed8814c9d455de9/mongo/options/clientoptions.go#L788C25-L788C39)

### System details

- OS: Linux 5.14.0-284.30.1.el9_2.x86_64
- Latest Cosmos Server

**Comment by azukaar** [comment]
The message seems to say that you used `/cosmos/` and not `/cosmos` in your URI?

**Comment by SSTentacleSS** [comment]
Thanks for your quick response!

**Comment by azukaar** [comment]
> Thanks for your quick response!

Meant to give you a better response than that ahah
thanks for the well written ticket the change will be tested and  introduced in 0.12 (prob next end of week)
thanks again

**Comment by azukaar** [comment]
fixed in 0.12
